php-frontend/src/test/java/org/sonar/php/cfg/ControlFlowGraphTest.java

```diff
@@ -46,133 +45,128 @@
  * -- 'pred' is a bracketed array of expected predecessor ids
  *
  * 2. each basic block must contain the 'block' function call as the first statement
- *
- * 3. the metadata string is constructed with the {@link BlockString}
  */
 public class ControlFlowGraphTest extends PHPTreeModelTest {
 
-  /**
-   * The predecessors are constructed based on the successors, so we should not test them all the time
-   */
   @Test
-  public void if_then_test_predecessors() {
-    ControlFlowGraph cfg = build(functionTree(
-      BlockString.withId(0).withSuccessorIds(1, 2).withPredecessorIds().build() +
-        "$x = 1;" +
-        "1 + 1;" +
+  public void test_parsing_for_start_block() {
+    // this is a test for the ExpectedStructureParser
+    ControlFlowGraph cfg = cfgForFunctionBody(
+      "block( [ id => 0, succ => [1,2], pred => [] ] );" +
         "if (a) {" +
-        BlockString.withId(1).withSuccessorIds(2).withPredecessorIds(0).build() +
+        "  block( [ id => 1, succ => [2], pred => [0] ] );" +
         "}" +
-        BlockString.withId(2).withSuccessorIds("END").withPredecessorIds(0, 1).build()));
+        "block( [ id => 2, succ => [END], pred => [0,1] ] );");
 
-    assertThat(cfg.blocks()).hasSize(4);
+    Map<String, ExpectedBlockStructure> expected = ExpectedStructureParser.parse(cfg.blocks());
+    assertThat(cfg.start()).isEqualTo(expected.get("0").actualBlock());
```

Comment:
do you think that this assertion is enough? what do we want to test with this test?